File: Object-C/BlackFireIM/Example/ChatDetail/Controller/BFWinkMessageCellData.m
//
//  BFCustomMessageCellData.m
//  BlackFireIM
//
//  Created by benny wang on 2021/4/13.
//

#import "BFWinkMessageCellData.h"


@implementation BFWinkMessageCellData

- (instancetype)init
{
    self = [super init];
    if (self) {
    }
    return self;
}

- (MSBusinessElem *)businessElem
{
    return (MSBusinessElem *)self.elem;
}

- (CGSize)contentSize
{
   return CGSizeMake(150, 150);
}

- (CGFloat)heightOfWidth:(CGFloat)width
{
    CGFloat height = 0;
    if (self.showName) {
        height += 25;
    }
    CGSize containerSize = [self contentSize];
    height += containerSize.height;
    if (self.direction == MsgDirectionOutgoing) {
        height += 20;
    }
    height += 5 + 5;
    return height;
}

- (NSString *)reuseId
{
    return @"TWinkMessageCell";
}

@end

File: Swift/MSIM-Swift/ChatDetail/C2C/BFWinkMessageCellData.swift
//
//  BFWinkMessageCellData.swift
//  MSIM-Swift
//
//  Created by benny wang on 2021/9/1.
//

import UIKit
import MSIMSDK


class BFWinkMessageCellData: MSMessageCellData {


    override func contentSize() -> CGSize {
        return CGSize(width: 150, height: 150)
    }
    
    override func heightOfWidth(width: CGFloat) -> CGFloat {
        var height: CGFloat = 0
        if showName {
            height += 25
        }
        let containerSize = self.contentSize()
        height += containerSize.height
        if direction == .outGoing {
            height += 20
        }
        height += 5 + 5
        return height
    }
    
    override var reUseId: String {
        return "TWinkMessageCell"
    }
}
